### ISSUE TITLE
Improve documentation on infrastructure requirements of the observability support

### ISSUE THREAD

**Issue by cezarysanecki** [spring-projects-experimental/spring-modulith#148]
Hi,

I've tried to start Spring Boot app using Modulith Observability module. It is very simple app just to check how works this piece of your software. Unfortunately Spring context is not starting. It is missing bean 'io.micrometer.tracing.Tracer'. If I register it as NOOP (did not find any concrete implementation) than app will start, but actuator does not contain my app as described [here in 7.1](https://docs.spring.io/spring-modulith/docs/0.2.0/reference/html/#observability).
```
@Bean
Tracer tracer() {
    return Tracer.NOOP;
}
```
```
{
   "_links":{
      "self":{
         "href":"http://localhost:8080/actuator",
         "templated":false
      },
      "health":{
         "href":"http://localhost:8080/actuator/health",
         "templated":false
      },
      "health-path":{
         "href":"http://localhost:8080/actuator/health/{*path}",
         "templated":true
      }
      (missing app) 
   }
}
```
I used Spring Boot 3.0.3 and Modulith 0.4.1. I added required libs from documentation. Also added io.micrometer.micrometer-tracing. If you need more data don't hesitate to contact me.

Thank you in advance and many thanks for your work.

Best regards.

**Comment by odrotbohm** [comment]
There are a couple of things at play here: actuator and observability support are separate things here. In other words, adding observability support will not expose the actuator endpoint. To activate the latte, make sure you follow the instructions listed [here](https://docs.spring.io/spring-modulith/docs/current/reference/html/#observability.actuator).

The observability support needs a `Tracer` instance, which is usually created by some Spring Boot auto-configuration triggered by the presence of some tracing infrastructure libraries present on the classpath. An example of such a setup using OTel and Zipkin can be found [here](https://github.com/odrotbohm/spring-restbucks/blob/b320e7052530072b50e7d553160399a421339c4c/server/pom.xml#L222-L245). I'll go ahead and use this ticket to add a bit more information regading those setup steps in our reference documentation.

**Comment by cezarysanecki** [comment]
Excuse me, but I have not seen message. Thanks for explanation and for clarifying doc. I will try this solution.

**Comment by odrotbohm** [comment]
No worries, I've added [a bit more information](https://docs.spring.io/spring-modulith/docs/0.5.0-SNAPSHOT/reference/html/#production-ready) on the additionally necessary dependencies to be declared in the documentation. To be shipped with 0.5.
